diff --git a/src/cli-sdk/src/commands/cache.ts b/src/cli-sdk/src/commands/cache.ts
index ba524cc6..6bed2038 100644
--- a/src/cli-sdk/src/commands/cache.ts
+++ b/src/cli-sdk/src/commands/cache.ts
@@ -27,9 +27,6 @@ export class CacheView extends ViewClass {
   stdout(...args: unknown[]) {
     stdout(...args)
   }
-  error(e: unknown) {
-    throw e
-  }
 }
 
 export const views: Views<void | CacheMap> = {
diff --git a/src/cli-sdk/src/config/index.ts b/src/cli-sdk/src/config/index.ts
index 77f246ce..96c0e7b5 100644
--- a/src/cli-sdk/src/config/index.ts
+++ b/src/cli-sdk/src/config/index.ts
@@ -26,9 +26,9 @@ import { PackageInfoClient } from '@vltpkg/package-info'
 import { PackageJson } from '@vltpkg/package-json'
 import { Monorepo } from '@vltpkg/workspaces'
 import { XDG } from '@vltpkg/xdg'
+import type { Jack, OptionsResults, Unwrap } from 'jackspeak'
 import { readFileSync, rmSync, writeFileSync } from 'node:fs'
 import { lstat, mkdir, readFile, writeFile } from 'node:fs/promises'
-import type { Jack, OptionsResults, Unwrap } from 'jackspeak'
 import { homedir } from 'node:os'
 import { dirname, resolve } from 'node:path'
 import { PathScurry } from 'path-scurry'
@@ -57,6 +57,8 @@ export {
   type Commands,
 }
 
+const kCustomInspect = Symbol.for('nodejs.util.inspect.custom')
+
 export type RecordPairs = Record<string, unknown>
 export type RecordString = Record<string, string>
 export type ConfigFiles = Record<string, ConfigFileData>
@@ -251,6 +253,17 @@ export class Config {
     )
     this.#options = Object.assign(options, {
       packageInfo: new PackageInfoClient(options),
+      [kCustomInspect]() {
+        return Object.fromEntries(
+          Object.entries(options).filter(
+            ([k]) =>
+              k !== 'monorepo' &&
+              k !== 'scurry' &&
+              k !== 'packageJson' &&
+              k !== 'packageInfo',
+          ),
+        )
+      },
     })
     return this.#options
   }
